aiohttp/http_parser.py

```diff
@@ -65,9 +67,56 @@
 METHRE: Final[Pattern[str]] = re.compile(r"[!#$%&'*+\-.^_`|~0-9A-Za-z]+")
 VERSRE: Final[Pattern[str]] = re.compile(r"HTTP/(\d).(\d)")
 HDRRE: Final[Pattern[bytes]] = re.compile(rb"[\x00-\x1F\x7F()<>@,;:\[\]={} \t\"\\]")
+QUOTEHDRRE = re.compile(r'(".*?(?:[^\\]"))[ \t]*(?:,|$)')
 HEXDIGIT = re.compile(rb"[0-9a-fA-F]+")
 
 
+class HeadersDictProxy(Mapping[str, str]):
```

Comment:
Have you compared performance compared to inheriting multidict and overriding the behaviors of storing the data and outputting the combined values?

Reply:
Implementation details are not a concern yet, so will come back to that. First, I want to get consensus that this is the correct approach and should be changed in v4, given that it is likely to cause backwards-compatibility breakages for atleast a small proportion of users.